A79-non-per-call-metrics-architecture.md

```diff
@@ -0,0 +1,598 @@
+of using an `experimental` namespace was considered, but not preferred since
+de-experimentalization of the metric would result in disruption of the
+corresponding metrics data (timeseries).
+
+New metrics should start as experimental. Once they are stable, they can
+optionally be made on-by-default. The distinction between on-by-default and
+off-by-default metrics allows gRPC to support metrics that may be desirable in
+some cases but too expensive to collect by default.
+
+### Changes to gRPC OpenTelemetry Plugin
+
+Optional Labels - Users should be able to configure gRPC OpenTelemetry plugins
+to record the additional optional labels. One suggested approach of implementing
+this is for the plugin instance to store a bit set for each instrument where
+each bit denotes whether the plugin should record the label in the corresponding
+position of the set of optional labels.
+
+API to enable/disable metrics - Users should be able to override the default
+behavior.
+
+Channel Scoped OpenTelemetry Plugins - API should be provided based on the
+approach chosen from the
+[Scoping Stats Plugins for Channels](#scoping-stats-plugins-for-channels)
+section.
+
+#### C++
+
+The following API will be added to `OpenTelemetryPluginBuilder` introduced in
+[A66: OpenTelemetry Metrics](A66-otel-stats.md).
+
+```c++
+class OpenTelemetryPluginBuilder {
+  using ChannelScope = grpc_core::StatsPlugin::ChannelScope;
+  // Adds all entries in \a optional_label_keys to the set of optional labels
+  // recorded across all metrics.
+  OpenTelemetryPluginBuilderImpl& AddOptionalLabels(
+      absl::Span<const absl::string_view> optional_label_keys);
+  // Set scope filter to choose which channels are recorded by this plugin.
+  // Server-side recording remains unaffected.
+  OpenTelemetryPluginBuilder& SetChannelScope(
+      absl::AnyInvocable<bool(ChannelScope /*scope*/) const> scope_filter);
+  // Enable/Disable metrics.
+  OpenTelemetryPluginBuilder& EnableMetrics(
+      absl::flat_hash_set<absl::string_view> metrics);
+  OpenTelemetryPluginBuilder& DisableMetrics(
+      absl::flat_hash_set<absl::string_view> metrics);
+  OpenTelemetryPluginBuilder& DisableAllMetrics();
+};
+```
+
+#### Java
+
+The following APIs will be added to `OpenTelemetryModule` introduced in
+[A66: OpenTelemetry Metrics](A66-otel-stats.md)
+
+```java
+public final class OpenTelemetryModule {
+  
+  /** Register plugin globally. */
+  public void registerGlobal();
+  
+  /** Register plugin for specific channel. */
+  public void configureChannelBuilder(ManagedChannelBuilder builder);
+
+  public static class Builder {
+    /** 
+     * Adds optionalLabelKey to all the metrics that can provide value for the 
+     * optionalLabelKey. */
+    public void addOptionalLabel(String optionalLabelKey);
+
+    /** 
+     * Enables metrics specified in the set along with metrics that are enabled 
+     * by default.
+     */
+    public Builder enableMetrics(Set<String> enableMetrics);
```

Comment:
@DNVindhya, users won't have a set for this already. We'll probably want to use `Collection` or `Iterable` instead.